### PR TITLE
Fix Quarkiverse template build and release

### PR DESCRIPTION
**Pull request by ia3andy**
Using only `verify` in mvn build and release won't let codestart tests pass.
Reason is that the tests are using the actual artifacts which should be installed in the local repo with `install`.

Note: `install` is the goal just after `verify`

### CODE REVIEW

**Comment by ia3andy** [comment]
@FroMage this is the PR to merge if your Renarde release passes.

**Comment by ia3andy** [comment]
@gastaldi no this is for all Quarkiverse extensions with a codestart

**Comment by gastaldi** [comment]
> > Ah I see. Are we already generating a codestart project next to the extension?
> 
> wdym?

You said this change is for Quarkiverse extensions with a codestart, I wonder if `quarkus create extension foobar`, for example, created a `codestart/` module next to the other modules

**Comment by ia3andy** [comment]
Ok it works: https://github.com/quarkiverse/quarkus-quinoa/actions/runs/2690774954

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
       - name: Maven release ${{steps.metadata.outputs.current-version}}
         run: |
           git checkout -b release
-          mvn -B release:prepare -Prelease -DreleaseVersion=${{steps.metadata.outputs.current-version}} -DdevelopmentVersion=${{steps.metadata.outputs.next-version}}
+          mvn -B release:prepare -Prelease -DpreparationGoals="clean install" -DreleaseVersion=${{steps.metadata.outputs.current-version}} -DdevelopmentVersion=${{steps.metadata.outputs.next-version}}
```
Comment (gastaldi) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: Can we add this to the project's pom.xml?

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: That might be possible but is it worth the effort? Do we release in some other place?

Comment (gastaldi) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: Actually here: https://github.com/quarkiverse/quarkiverse-parent/blob/main/pom.xml#L105, as in https://github.com/forge/core/blob/master/pom.xml#L244

Comment (gastaldi) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: https://github.com/quarkiverse/quarkiverse-parent/pull/56

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: Ok let's wait for the Quarkiverse parent release and I'll update this PR (parent version + build)  then 👍

Comment (gsmet) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: @ia3andy could you rebase this PR with the latest parent? Thanks!

Comment (aalmiray) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: FWIW JReleaser 1.2.0 comes out by the end of August. We've adopted a 2 month release cadence for stable releases to help with planning https://github.com/jreleaser/jreleaser/discussions/873

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/release.yml:50]: I've rebased, but I am not sure if we should merge this or not... Maybe it's better to have this until we have the actual fix ready?

**Comment by ia3andy** [comment]
@gsmet I think we should merge this for `2.12.Final` without this new Quarkiverse extensions with codestarts won't pass CI.

Anyway it will all change, when @gastaldi add JReleaser.

**Review by gsmet — APPROVED** [review]

**Pull request merged**